metasynth/distribution/categorical.py

```diff
@@ -42,33 +42,52 @@ def __init__(self, labels: Union[npt.NDArray[np.str_], list[str]],
     def _fit(cls, values: pl.Series):
         labels, counts = np.unique(values, return_counts=True)
         probs = counts/np.sum(counts)
-        return cls(labels.astype(str), probs)
+        return cls(labels, probs)
 
     def _param_dict(self):
         return {"labels": self.labels, "probs": self.probs}
 
     @classmethod
     def _param_schema(cls):
         return {
-            "labels": {"type": "array", "items": {"type": "string"}, "uniqueItems": True},
+            "labels": {"type": "array", "uniqueItems": True},
             "probs": {"type": "array", "items": {"type": "number"}},
         }
 
     def draw(self):
-        return str(np.random.choice(self.labels, p=self.probs))
+        return np.random.choice(self.labels, p=self.probs)
 
     def information_criterion(self,
                               values: Union[pd.Series, pl.Series, npt.NDArray[np.str_]]
                               ) -> float:
-        values_array = np.array(values, dtype=str)
-        labels, counts = np.unique(values_array, return_counts=True)
+        series = self._to_series(values)
+        labels, counts = np.unique(series, return_counts=True)
         log_lik = 0.0
         pdict = dict(zip(self.labels, self.probs))
+        if len(self.labels) > 1 and isinstance(self.labels[0], (int, np.int_)):
+            # Do add-one smoothing, assume count of values is equal to one in initial fit.
+            alpha = 1
+            num_event = len(series)
+            num_tot_cat = np.max(self.labels)-np.min(self.labels)+1
+            num_zero_cat = num_tot_cat - len(self.labels)
+            theta = (self.probs*num_event+alpha)/(num_event+num_tot_cat*alpha)
+            theta_zero = alpha/(num_event+num_tot_cat*alpha)
+            theta_sum = np.sum(theta) + num_zero_cat*theta_zero
+            theta_dict = dict(zip(self.labels, theta))
+            for lab, count in zip(labels, counts):
+                cur_theta = theta_dict.get(lab, theta_zero)
+                cur_prob = cur_theta/theta_sum
+                log_lik += count * np.log(cur_prob)
+            return 2*(2*len(self.probs)-1) - 2*log_lik
```

Comment:
This should be actual AIC, i.e.:

```suggestion
            return 2*(len(self.probs)-1) - 2*log_lik
```

Reply:
I kind of agree, but can't we see both the label and the probability as a parameter? If you compare to a uniform distribution [a, b], which has according to our definitions two parameters, then the multinoulli would have 2? parameters as well, except that in reality it has effectively [a, p_a, b, p_b] parameters, where you can have p_a == p_b, but don't have to adhere to this. Personally, I think this should be classified as 4 parameters, at least in the small count/label limit. In the limit of large n, this factor has limited influence as well. Another consideration that I think generally multinoulli a little bit more discouraged than perhaps normally, since we don't want too many parameters in our distributions.